app/models/shop/stock.rb

```diff
@@ -0,0 +1,14 @@
+class Shop::Stock < ApplicationRecord
+  belongs_to :book
+  belongs_to :shop
+
+  validates :value, numericality: { greater_than_or_equal_to: 0 }
+
+  scope :with_books, -> (book_ids) { where(book_id: book_ids) }
+
+  def initialize(attrs = {})
+    super(attrs)
+    self.value = 0 if value.nil?
```

Comment:
why not `t.integer "value", default: 0`?

Reply:
I think spreading logic between app and database isn't a good idea.
What will I do if later default will be 1? Update database? And if I will have a billion rows? 
That's why I'm setting up the default value in the initializer.

Reply:
make sense, but then there is no point to use default values at all if they always can be changed.